src/Test/TestCombatMods.java

```diff
@@ -0,0 +1,131 @@
+
+    testMap = new MapMaster(cos, MapLibrary.getByName("Firing Range"));
+    for( Commander co : cos )
+    {
+      co.myView = new MapWindow(testMap, co);
+    }
+  }
+
+  @Override
+  public boolean runTest()
+  {
+    setupTest();
+    cinder.registerForEvents();
+    venge.registerForEvents();
+
+    boolean testPassed = true;
+    testPassed &= validate(testBasicMod(), "  Basic combat mod test failed.");
+    testPassed &= validate(testRetribution(), "  Venge's Retribution combat mod test failed.");
+    
+    cinder.unregister();
+    venge.unregister();
+    return testPassed;
+  }
+
+  /** Test that combat modifiers are applied both in defense and offense. */
+  private boolean testBasicMod()
+  {
+    // Add our test subjects
+    Unit infActive = addUnit(testMap, cinder, UnitEnum.INFANTRY, 7, 3);
+    infActive.initTurn(testMap); // Make sure he is ready to move.
+    Unit infPassive = addUnit(testMap, cinder, UnitEnum.INFANTRY, 7, 5);
+    
+    // We need a victim and an angry man to avenge him
+    Unit bait = addUnit(testMap, venge, UnitEnum.APC, 7, 4);
+    Unit meaty = addUnit(testMap, venge, UnitEnum.MD_TANK, 8, 4);
+    
+    // Poke the bear...
+    performGameAction(new GameAction.AttackAction(testMap, infActive, Utils.findShortestPath(infActive, 7, 3, testMap), 7, 4), testMap);
+
+    // See how much damage meaty can do to our two contestants on offense...
+    BattleSummary vengeful = CombatEngine.simulateBattleResults(meaty, infActive, testMap, 8, 3);
+    BattleSummary normal = CombatEngine.simulateBattleResults(meaty, infPassive, testMap, 8, 5);
+    // ...and defense
+    BattleSummary vengefulCounter = CombatEngine.simulateBattleResults(infActive, meaty, testMap, 8, 3);
+    BattleSummary normalCounter = CombatEngine.simulateBattleResults(infPassive, meaty, testMap, 8, 5);
+    
+    // Check that Venge's passive ability works on both attack and defense
+    boolean testPassed = validate(vengeful.defenderHPLoss > normal.defenderHPLoss, "    Being angry didn't help Venge attack extra hard.");
+    testPassed &= validate(vengefulCounter.attackerHPLoss > normalCounter.attackerHPLoss, "    Being angry didn't help Venge defend extra hard.");
+
+    // Clean up
+    testMap.removeUnit(infActive);
+    testMap.removeUnit(infPassive);
+    testMap.removeUnit(bait);
+    testMap.removeUnit(meaty);
+
+    return testPassed;
+  }
+
+  /** Test that combat works as a black box, but that Venge gets his way all the same. */
+  private boolean testRetribution()
+  {
+    // Add our test subjects
+    Unit infA = addUnit(testMap, cinder, UnitEnum.INFANTRY, 7, 3);
+    Unit infB = addUnit(testMap, venge, UnitEnum.INFANTRY, 7, 5);
+    
+    // Check our damage for each first strike pre-power...
+    BattleSummary normalAB = CombatEngine.simulateBattleResults(infA, infB, testMap, 7, 4);
+    BattleSummary normalBA = CombatEngine.simulateBattleResults(infB, infA, testMap, 7, 4);
+
+    venge.modifyAbilityPower(42); // juice up
```

Comment:
Maybe do `modifyAbilityPower( venge.getReadyAbilities().get(1).getCost() )` to ensure this test case survives any future balance changes?

Reply:
And also maybe check the name of the ability to ensure are using the one we think we are. That's less likely to change though, I would think.

Reply:
`getReadyAbilities()` doesn't help if you don't already have any CO energy, it'll just return empty.
...and I don't think any power will ever cost 42+ stars at base, haha.

Checking the name could work, but it's really unnecessary. If anything changes, you're gonna try to run the game to test it.... and the test log will tell you.